### PR TITLE
OCPBUGS#33700:Added routes note for soure IP address in IController doc

### PR DESCRIPTION
**Pull request by dfitzmau**
Version(s):
4.14+

Issue:
[OCPBUGS-33700](https://issues.redhat.com/browse/OCPBUGS-33700)

Link to docs preview:
* [Configuring the PROXY protocol for an Ingress Controller-Dedicated](https://78097--ocpdocs-pr.netlify.app/openshift-dedicated/latest/networking/ingress-operator.html#nw-ingress-controller-configuration-proxy-protocol_configuring-ingress)
* [Configuring the PROXY protocol for an Ingress Controller-ROSA](https://78097--ocpdocs-pr.netlify.app/openshift-rosa/latest/networking/ingress-operator.html#nw-ingress-controller-configuration-proxy-protocol_configuring-ingress)
* [Configuring the PROXY protocol for an Ingress Controller-Enterprise](https://78097--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/ingress-operator.html#nw-ingress-controller-configuration-proxy-protocol_configuring-ingress)

- [x] SME has approved this change.
- [x] QE has approved this change.

### CODE REVIEW

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 1]
```diff
@@ -6,9 +6,18 @@
 [id="nw-ingress-controller-configuration-proxy-protocol_{context}"]
 = Configuring the PROXY protocol for an Ingress Controller
 
-A cluster administrator can configure https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt[the PROXY protocol] when an Ingress Controller uses either the `HostNetwork` or `NodePortService` endpoint publishing strategy types. The PROXY protocol enables the load balancer to preserve the original client addresses for connections that the Ingress Controller receives. The original client addresses are useful for logging, filtering, and injecting HTTP headers. In the default configuration, the connections that the Ingress Controller receives only contain the source address that is associated with the load balancer.
```
Comment (Miciah) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:9]: Since [OCPBUGS-3559](https://issues.redhat.com/browse/OCPBUGS-3559) was fixed in OCP 4.12.0 and backported to OCP 4.11.40 as [OCPBUGS-33700](https://issues.redhat.com/browse/OCPBUGS-33700), PROXY protocol can also be enabled for the `Private` endpoint publishing strategy type.  Would you like me to file a separate bug report for that?

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 2]
```diff
@@ -6,9 +6,18 @@
...
+[IMPORTANT]
+====
+For a passthrough route configuration, client pods in an {product-title} cluster cannot access their original client source IP address from within the cluster. Ensure that you configure Ingress access logging for your Ingress Controller so that you can view what source IP addresses client pods can access from within your cluster.
+
+For re-encrypt and edge routes, you can set the `X-Forwarded-For` value in the `httpHeaders` of the `ingresscontrollers.operator.openshift.io` resource to allow the client source IP address to reach client pods in a cluster.
+
+For more information about Ingress access logging, see the "Configuring Ingress access logging".
+====
+
+Configuring the PROXY protocol for an Ingress Controller is not supported in cloud deployments. This restriction is because when {product-title} runs in a cloud platform, and an IngressController specifies that a service load balancer should be used, the Ingress Operator configures the load balancer service and enables the PROXY protocol based on the platform requirement for preserving source addresses.
```
Comment (Miciah) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:20]: There may be some confusion resulting from the fact that the _default_ endpoint publishing strategy type on cloud platforms is `LoadBalancerService`; however, the other endpoint publishing strategies can still be specified explicitly, and then it is possible to enable PROXY protocol.  So it would be more accurate to say that PROXY protocol cannot be configured _when using `LoadBalancerService`_.  
```suggestion
Configuring the PROXY protocol for an Ingress Controller is not supported when using the `LoadBalancerService` endpoint publishing strategy type. This restriction is because when {product-title} runs in a cloud platform, and an IngressController specifies that a service load balancer should be used, the Ingress Operator configures the load balancer service and enables the PROXY protocol based on the platform requirement for preserving source addresses.
```

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 3]
```diff
@@ -6,9 +6,18 @@
...
 
-This feature is not supported in cloud deployments. This restriction is because when {product-title} runs in a cloud platform, and an IngressController specifies that a service load balancer should be used, the Ingress Operator configures the load balancer service and enables the PROXY protocol based on the platform requirement for preserving source addresses.
+[IMPORTANT]
+====
+For a passthrough route configuration, client pods in an {product-title} cluster cannot access their original client source IP address from within the cluster. Ensure that you configure Ingress access logging for your Ingress Controller so that you can view what source IP addresses client pods can access from within your cluster.
```
Comment (Miciah) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:13]: A few corrections:

- This issue doesn't only apply to client pods; it applies to all client connections.  
- It isn't the client that would read the source IP address; it's the server pod.  
- Using term "access" could be misunderstood to imply that the pod cannot connect to the IP address; maybe "observe" would be clearer?
- As there is some cost to recording and storing access logs, I believe it would be better to instruct the cluster-admin to enable access logs only if needed.
```suggestion
For a passthrough route configuration, servers in an {product-title} cluster cannot observe the original client source IP address. If you need to know the original client source IP address, ensure that you configure Ingress access logging for your Ingress Controller so that you can view the client source IP addresses.  
```

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 4]
```diff
@@ -6,9 +6,18 @@
...
+
+For re-encrypt and edge routes, you can set the `X-Forwarded-For` value in the `httpHeaders` of the `ingresscontrollers.operator.openshift.io` resource to allow the client source IP address to reach client pods in a cluster.
```
Comment (Miciah) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:15]: OpenShift router sets the `forwarded` and `x-forwarded-for` headers itself; in the general case, there should be no need to configure `httpHeaders` actions to set these headers.
```suggestion
For re-encrypt and edge routes, {product-title} router sets the `Forwarded` and `X-Forwarded-For` headers to enable servers to check the client source IP address.
```

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 2 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:20]: Thanks. Updated.

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 3 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:13]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 4 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:15]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 1 above)
Comment (Miciah) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:9]: I filed [OCPBUGS-36292](https://issues.redhat.com/browse/OCPBUGS-36292) for this issue.

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 5]
```diff
@@ -6,9 +6,18 @@
...
 
-This feature is not supported in cloud deployments. This restriction is because when {product-title} runs in a cloud platform, and an IngressController specifies that a service load balancer should be used, the Ingress Operator configures the load balancer service and enables the PROXY protocol based on the platform requirement for preserving source addresses.
+[IMPORTANT]
+====
+For a passthrough route configuration, servers in an {product-title} cluster cannot observe the original client source IP address. If you need to know the original client source IP address, ensure that you configure Ingress access logging for your Ingress Controller, so that you can view the client source IP addresses.
+
+For re-encrypt and edge routes, the {product-title} router sets the `Forwarded` and `X-Forwarded-For` headers to enable servers to check the client source IP address.
+
+For more information about Ingress access logging, see the "Configuring Ingress access logging".
```
Comment (Miciah) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:17]: Should this reference be a xref?

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 6]
```diff
@@ -33,7 +42,7 @@ $ oc -n openshift-ingress-operator edit ingresscontroller/default
 
 . Set the PROXY configuration:
 +
-* If your Ingress Controller uses the hostNetwork endpoint publishing strategy type, set the `spec.endpointPublishingStrategy.hostNetwork.protocol` subfield to `PROXY`:
+* If your Ingress Controller uses the `hostNetwork` endpoint publishing strategy type, set the `spec.endpointPublishingStrategy.hostNetwork.protocol` subfield to `PROXY`:
```
Comment (Miciah) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:45]: `HostNetwork` should be capitalized.
```suggestion
* If your Ingress Controller uses the `HostNetwork` endpoint publishing strategy type, set the `spec.endpointPublishingStrategy.hostNetwork.protocol` subfield to `PROXY`:
```

**Comment by Miciah** [comment]
Sorry if you see duplicate comments or get duplicate notifications.  I added comments on https://github.com/openshift/openshift-docs/commit/431fa1f8624890c94fe444233b4532001dcd49b3 and noticed they weren't showing up here, and so I added the same comments on https://github.com/openshift/openshift-docs/pull/78097/commits/431fa1f8624890c94fe444233b4532001dcd49b3.  

The new text looks good.  My only comments were a suggestion to add a xref and a tiny fix for some existing text that your PR happens to touch.  
/lgtm

**Comment by Miciah** [comment]
In my previous comment, the first link has `commit/431fa1f8624890c94fe444233b4532001dcd49b3` where the second link has `pull/78097/commits/431fa1f8624890c94fe444233b4532001dcd49b3`.  The rendered comment has them converted to the same link.  GitHub is confusing.

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 5 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:17]: Unfortunetly, xref-support at the module-level is not possible. Instead, it's at the assembly level (higher-level directory).

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 6 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:47]: Updated.

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 7]
```diff
@@ -1,23 +1,34 @@
 // Module included in the following assemblies:
 //
-// * ingress/configure-ingress-operator.adoc
+// * ingress/ingress-operator.adoc
```
Comment (Miciah) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:3]: Should this be `networking/ingress-operator.adoc`?

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 5 above)
Comment (Miciah) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:24]: One other thing: Should the "the" be removed?
```suggestion
For more information about Ingress access logging, see "Configuring Ingress access logging".
```
Or:
```suggestion
For more information about Ingress access logging, see the "Configuring Ingress access logging" section.
```

Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:24]: Updated

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 7 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc]: Correct. Updated.

Screenshot from 2024-06-28 11-38-22

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 8]
```diff
@@ -1,23 +1,34 @@
...
+====
+
+The PROXY protocol enables the load balancer to preserve the original client addresses for connections that the Ingress Controller receives. The original client addresses are useful for logging, filtering, and injecting HTTP headers. In the default configuration, the connections that the Ingress Controller receives only contain the source IP address that is associated with the load balancer.
 
 [IMPORTANT]
 ====
-You must configure both {product-title} and the external load balancer to either use the PROXY protocol or to use TCP.
+For a passthrough route configuration, servers in an {product-title} cluster cannot observe the original client source IP address. If you need to know the original client source IP address, ensure that you configure Ingress access logging for your Ingress Controller, so that you can view the client source IP addresses.
+
+For re-encrypt and edge routes, the {product-title} router sets the `Forwarded` and `X-Forwarded-For` headers to enable servers to check the client source IP address.
```
[the remark this answers is not in the document]
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc]: Thanks. Updated!

**Comment by Miciah** [comment]
Thanks!
/lgtm

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 9]
```diff
@@ -1,23 +1,34 @@
...
+For more information about Ingress access logging, see "Configuring Ingress access logging".
 ====
 
-[WARNING]
+Configuring the PROXY protocol for an Ingress Controller is not supported when using the `LoadBalancerService` endpoint publishing strategy type. This restriction is because when {product-title} runs in a cloud platform, and an IngressController specifies that a service load balancer should be used, the Ingress Operator configures the load balancer service and enables the PROXY protocol based on the platform requirement for preserving source addresses.
+
+[IMPORTANT]
 ====
-The PROXY protocol is unsupported for the default Ingress Controller with installer-provisioned clusters on non-cloud platforms that use a Keepalived Ingress VIP.
+You must configure both {product-title} and the external load balancer to either use the PROXY protocol or to use TCP.
```
Comment (maxwelldb) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:31]:
```suggestion
You must configure both {product-title} and the external load balancer to use either the PROXY protocol or TCP.
```

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 10]
```diff
@@ -1,23 +1,34 @@
...
+A cluster administrator can configure link:https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt[the PROXY protocol] when an Ingress Controller uses either the `HostNetwork` or `NodePortService` endpoint publishing strategy types.
 
-This feature is not supported in cloud deployments. This restriction is because when {product-title} runs in a cloud platform, and an IngressController specifies that a service load balancer should be used, the Ingress Operator configures the load balancer service and enables the PROXY protocol based on the platform requirement for preserving source addresses.
+[WARNING]
+====
+The PROXY protocol is unsupported for the default Ingress Controller with installer-provisioned clusters on non-cloud platforms that use a Keepalived Ingress VIP.
```
Comment (maxwelldb) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:13]: Consider expanding "VIP" on first use.

Comment (maxwelldb) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:13]: Could rewrite to be active, e.g. "The default Ingress Controller with installer-provisioned clusters on non-cloud platforms that use a Keepalived Ingress VIP do not support the PROXY protocol."

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 11]
```diff
@@ -1,23 +1,34 @@
...
+====
+
+The PROXY protocol enables the load balancer to preserve the original client addresses for connections that the Ingress Controller receives. The original client addresses are useful for logging, filtering, and injecting HTTP headers. In the default configuration, the connections that the Ingress Controller receives only contain the source IP address that is associated with the load balancer.
 
 [IMPORTANT]
 ====
-You must configure both {product-title} and the external load balancer to either use the PROXY protocol or to use TCP.
+For a passthrough route configuration, servers in an {product-title} cluster cannot observe the original client source IP address. If you need to know the original client source IP address, ensure that you configure Ingress access logging for your Ingress Controller, so that you can view the client source IP addresses.
```
Comment (maxwelldb) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:20]:
```suggestion
For a passthrough route configuration, servers in an {product-title} cluster cannot observe the original client source IP address. If you need to know the original client source IP address, configure Ingress access logging for your Ingress Controller so that you can view the client source IP addresses.
```

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 12]
```diff
@@ -1,23 +1,34 @@
...
+
+For re-encrypt and edge routes, the {product-title} router sets the `Forwarded` and `X-Forwarded-For` headers to enable the application workloads to check the client source IP address.
```
Comment (maxwelldb) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:22]: re: "enable" in general, consider a different verb. ISG prefers to reserve that for cases where you are enabling/disabling a functionality.

**Review by maxwelldb — COMMENTED** [review]
Left some nit-level comments.

**Comment by lihongan** [comment]
@ShudiLi could you please help take a look? thanks.
cc @melvinjoseph86 @rhamini3

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 13]
```diff
@@ -44,7 +55,7 @@ $ oc -n openshift-ingress-operator edit ingresscontroller/default
         protocol: PROXY
       type: HostNetwork
 ----
-* If your Ingress Controller uses the NodePortService endpoint publishing strategy type, set the `spec.endpointPublishingStrategy.nodePort.protocol` subfield to `PROXY`:
+* If your Ingress Controller uses the `NodePortService` endpoint publishing strategy type, set the `spec.endpointPublishingStrategy.nodePort.protocol` subfield to `PROXY`:
```
Comment (ShudiLi) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:58]: can add the following for supporting the Private endpoint publishing strategy type:
If your Ingress Controller uses the `Private` endpoint publishing strategy type, set the `spec.endpointPublishingStrategy.private.protocol` subfield to `PROXY`:
.Sample `private` configuration to `PROXY`
spec:
  endpointPublishingStrategy:
    type: Private
    private:
      protocol: PROXY

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 9 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:31]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 10 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:13]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 11 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:20]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 12 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:22]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 14]
```diff
@@ -1,23 +1,34 @@
...
 //
-// * ingress/configure-ingress-operator.adoc
+// * networking/ingress-operator.adoc
 
 :_mod-docs-content-type: PROCEDURE
 [id="nw-ingress-controller-configuration-proxy-protocol_{context}"]
 = Configuring the PROXY protocol for an Ingress Controller
 
-A cluster administrator can configure https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt[the PROXY protocol] when an Ingress Controller uses either the `HostNetwork` or `NodePortService` endpoint publishing strategy types. The PROXY protocol enables the load balancer to preserve the original client addresses for connections that the Ingress Controller receives. The original client addresses are useful for logging, filtering, and injecting HTTP headers. In the default configuration, the connections that the Ingress Controller receives only contain the source address that is associated with the load balancer.
+A cluster administrator can configure link:https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt[the PROXY protocol] when an Ingress Controller uses either the `HostNetwork` or `NodePortService` endpoint publishing strategy types.
```
[the remark this answers is not in the document]
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:9]: Thanks, @ShudiLi . Updated.

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 13 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:61]: :heavy_check_mark:

**Comment by ShudiLi** [comment]
/lgtm
thanks

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 15]
```diff
@@ -1,23 +1,34 @@
...
 //
-// * ingress/configure-ingress-operator.adoc
+// * networking/ingress-operator.adoc
 
 :_mod-docs-content-type: PROCEDURE
 [id="nw-ingress-controller-configuration-proxy-protocol_{context}"]
 = Configuring the PROXY protocol for an Ingress Controller
 
-A cluster administrator can configure https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt[the PROXY protocol] when an Ingress Controller uses either the `HostNetwork` or `NodePortService` endpoint publishing strategy types. The PROXY protocol enables the load balancer to preserve the original client addresses for connections that the Ingress Controller receives. The original client addresses are useful for logging, filtering, and injecting HTTP headers. In the default configuration, the connections that the Ingress Controller receives only contain the source address that is associated with the load balancer.
+A cluster administrator can configure link:https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt[the PROXY protocol] when an Ingress Controller uses either the `HostNetwork`, `NodePortService`, or 'Private' endpoint publishing strategy types.
```
Comment (skopacz1) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:9]: These should be backticks - Private doesn't render as code text in the preview:
```suggestion
A cluster administrator can configure link:https://www.haproxy.org/download/1.8/doc/proxy-protocol.txt[the PROXY protocol] when an Ingress Controller uses either the `HostNetwork`, `NodePortService`, or `Private` endpoint publishing strategy types.
```

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 16]
```diff
@@ -1,23 +1,34 @@
...
 
-This feature is not supported in cloud deployments. This restriction is because when {product-title} runs in a cloud platform, and an IngressController specifies that a service load balancer should be used, the Ingress Operator configures the load balancer service and enables the PROXY protocol based on the platform requirement for preserving source addresses.
+[WARNING]
+====
+The default Ingress Controller with installer-provisioned clusters on non-cloud platforms that use a Keepalived Ingress Virtual IP (VIP) do not support the PROXY protocol.
+====
+
+The PROXY protocol enables the load balancer to preserve the original client addresses for connections that the Ingress Controller receives. The original client addresses are useful for logging, filtering, and injecting HTTP headers. In the default configuration, the connections that the Ingress Controller receives only contain the source IP address that is associated with the load balancer.
```
Comment (skopacz1) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:16]: Nit, per the [ISG](https://www.ibm.com/docs/en/ibm-style?topic=grammar-adverbs-only) the word "only" should be as close to the phrase it modifies:
```suggestion
The PROXY protocol enables the load balancer to preserve the original client addresses for connections that the Ingress Controller receives. The original client addresses are useful for logging, filtering, and injecting HTTP headers. In the default configuration, the connections that the Ingress Controller receives contain only the source IP address that is associated with the load balancer.
```
If you feel that the original placement is the more accurate meaning, feel free to ignore this comment

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 17]
```diff
@@ -1,23 +1,34 @@
...
 
 [IMPORTANT]
 ====
-You must configure both {product-title} and the external load balancer to either use the PROXY protocol or to use TCP.
+For a passthrough route configuration, servers in an {product-title} cluster cannot observe the original client source IP address. If you need to know the original client source IP address, configure Ingress access logging for your Ingress Controller so that you can view the client source IP addresses.
```
Comment (skopacz1) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:20]: Very unfortunate - in the ROSA doc, "an {product-title} cluster" is incorrect because it renders to "an Red Hat OpenShift...".

Up to you how to resolve this, but I feel like the simplest option would be to remove the article and make cluster plural:
```suggestion
For a passthrough route configuration, servers in {product-title} clusters cannot observe the original client source IP address. If you need to know the original client source IP address, configure Ingress access logging for your Ingress Controller so that you can view the client source IP addresses.
```

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 18]
```diff
@@ -1,23 +1,34 @@
...
+
+For re-encrypt and edge routes, the {product-title} router sets the `Forwarded` and `X-Forwarded-For` headers so that application workloads check the client source IP address.
+
+For more information about Ingress access logging, see "Configuring Ingress access logging".
 ====
 
-[WARNING]
+Configuring the PROXY protocol for an Ingress Controller is not supported when using the `LoadBalancerService` endpoint publishing strategy type. This restriction is because when {product-title} runs in a cloud platform, and an IngressController specifies that a service load balancer should be used, the Ingress Operator configures the load balancer service and enables the PROXY protocol based on the platform requirement for preserving source addresses.
```
Comment (skopacz1) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:27]: Missing space here?
```suggestion
Configuring the PROXY protocol for an Ingress Controller is not supported when using the `LoadBalancerService` endpoint publishing strategy type. This restriction is because when {product-title} runs in a cloud platform, and an Ingress Controller specifies that a service load balancer should be used, the Ingress Operator configures the load balancer service and enables the PROXY protocol based on the platform requirement for preserving source addresses.
```

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 19]
```diff
@@ -33,25 +44,44 @@ $ oc -n openshift-ingress-operator edit ingresscontroller/default
 
 . Set the PROXY configuration:
 +
-* If your Ingress Controller uses the hostNetwork endpoint publishing strategy type, set the `spec.endpointPublishingStrategy.hostNetwork.protocol` subfield to `PROXY`:
+* If your Ingress Controller uses the `HostNetwork` endpoint publishing strategy type, set the `spec.endpointPublishingStrategy.hostNetwork.protocol` subfield to `PROXY`:
```
Comment (skopacz1) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:47]: Based on the source block below, should the H be capitalized?
```suggestion
* If your Ingress Controller uses the `hostNetwork` endpoint publishing strategy type, set the `spec.endpointPublishingStrategy.hostNetwork.protocol` subfield to `PROXY`:
```

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc
Comment (skopacz1) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:1]: This is out of scope to your work and you don't have to do this at all, but if you're feeling up for it: [this line](https://github.com/openshift/openshift-docs/pull/78097/files#diff-da5babc2dce52b79d99be94bdaf4e0890583b18c2e7d9ebf8d42f1cb5f2547a3R38) (step one of this procedure) could have "by running the following command" added to the end of it.

**Review by skopacz1 — COMMENTED** [review]
Left a few comments, let me know when they're addressed and I can proceed to merge!

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 15 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:9]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 16 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:16]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 17 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:20]: Well spotted!

:heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 18 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:27]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 19 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:47]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:1]: Thanks, valid. Updated. Might as well while I'm strolling through this part of the doc woods!

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc [hunk 20]
```diff
@@ -1,30 +1,41 @@
...
-The PROXY protocol is unsupported for the default Ingress Controller with installer-provisioned clusters on non-cloud platforms that use a Keepalived Ingress VIP.
+You must configure both {product-title} and the external load balancer to use either the PROXY protocol or TCP.
 ====
 
 .Prerequisites
 * You created an Ingress Controller.
 
 .Procedure
-. Edit the Ingress Controller resource:
+. Edit the Ingress Controller resourc by entering the following command in your CLI:
```
Comment (skopacz1) [inline modules/nw-ingress-controller-configuration-proxy-protocol.adoc:38]: One more nit: 
```suggestion
. Edit the Ingress Controller resource by entering the following command in your CLI:
```

**Review by skopacz1 — COMMENTED** [review]
Just one more nit

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 20 above)
Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc]: :heavy_check_mark:

File: modules/nw-ingress-controller-configuration-proxy-protocol.adoc (hunk 19 above)
Comment (ShudiLi) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc]: There are the two keywords in the below IC configuration file:
```
apiVersion: operator.openshift.io/v1
kind: IngressController
metadata:
  name: int1
  namespace: openshift-ingress-operator
spec:
  replicas: 1
  endpointPublishingStrategy:
    hostNetwork:
      protocol: PROXY
    type: HostNetwork
  domain: int1.shudi-4v702.qe.devcluster.openshift.com
```

Comment (dfitzmau) [reply modules/nw-ingress-controller-configuration-proxy-protocol.adoc:54]: Thanks, @ShudiLi
